Remove old `/_matrix/client/*/admin` endpoints (#8785)
matrix-org/synapse: UPGRADE.rst

          return {"localpart": localpart}

> Removal historically Synapse Admin API 
> --------------------------------------

Reviewer: ```suggestion
Removal historical Synapse Admin API 
------------------------------------
```

Historically, the Synapse Admin API has been accessible under:

* ``/_matrix/client/api/v1/admin``
* ``/_matrix/client/unstable/admin``
* ``/_matrix/client/r0/admin``
* ``/_synapse/admin/v1``

> These endpoints with ``/_matrix/client/*`` prefixes have been removed.
> From this point, the Admin API will only be accessible under:

Reviewer: I think it is nice to re-iterate the version from the header above:

```suggestion
The endpoints with ``/_matrix/client/*`` prefixes have been removed as of v1.24.0.
The Admin API is now only accessible under:
```

* ``/_synapse/admin/v1``

> The only exception is:
> 
> * ``/_matrix/client/r0/admin/whois/<user_id>``
> * ``/_synapse/admin/v1/whois/<user_id>``

Reviewer: I don't think we need to provide the full paths again (since there is potentially multiple paths this is available at on the client-server API):

```suggestion
The only exception is the `/admin/whois` endpoint, which is
`also available via the client-server API <https://matrix.org/docs/spec/client_server/r0.6.1#get-matrix-client-r0-admin-whois-userid>`_.
```

> This change has been announced with Synapse 1.20.0 (2020-09-22) and makes it easier
> for homeserver admins to lock down external access to the Admin API endpoints.

Reviewer: ```suggestion
The deprecation of the old endpoints was announced with Synapse 1.20.0 (released
on 2020-09-22) and makes it easier for homeserver admins to lock down external
access to the Admin API endpoints.
```
